horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/ActionServiceTest.java

```diff
@@ -60,4 +83,75 @@ public void testAddGlobalAction() {
    }
+
+   /**
+    * @param testInfo
+    */
+   @org.junit.jupiter.api.Test
+   public void testMessageBusAction(TestInfo testInfo) {
+      Test test = createTest(createExampleTest(getTestName(testInfo)));
+
+      String channel = "NOTACHANNEL";
+      String token = "BADTOKEN";
+
+      // Build an Action JSON directly because Action DTO serialization will
+      // mask the secrets. (Arguably, since we want to provoke an error here,
+      // the masked "****" value would work just as well, but this code runs
+      // successfully if the channel and token are real.)
+      ObjectMapper mapper = new ObjectMapper();
+      ObjectNode config = mapper.createObjectNode()
+            .put("channel", channel).put("formatter", "testToSlack");
+      ObjectNode secrets = mapper.createObjectNode().put("token", token);
+      ObjectNode action = mapper.createObjectNode()
+            .put("id", -1)
+            .put("event", MessageBusChannels.TEST_NEW.name())
+            .put("type", SlackChannelMessageAction.TYPE_SLACK_MESSAGE)
+            .put("testId", test.id)
+            .put("active", true)
+            .put("runAlways", true);
+
+      // `set` returns a JsonNode, which can't be `set` or `put`, so we do
+      // these outside the chain.
+      action.set("config", config);
+      action.set("secrets", secrets);
+      Action r = jsonRequest().auth().oauth2(getAdminToken())
+            .contentType(ContentType.JSON).body(action)
+            .post("/api/action")
+            .then().statusCode(200)
+            .contentType(ContentType.JSON).extract().body().as(Action.class);
+
+      // send new test to action service
+      actionService.onNewTest(test);
+
+      log.infof("Waiting for action to run");
+      int errors = 0;
+      for (int i = 0; i < 10; i++) {
+         try {
+            Thread.sleep(1000);
+         } catch (InterruptedException e) {
+         }
+
+         // Should these action errors be logged under dummyTest.id? The
+         // ActionServiceImpl handler seems to always log under its argument
+         // value, -1: but I still can't find them.
+         List<ActionLog> logs = jsonRequest().auth().oauth2(getTesterToken())
+               .queryParam("level", PersistentLogDAO.ERROR).get("/api/log/action/-1")
+               .then().statusCode(200)
+               .contentType(ContentType.JSON).extract().body().jsonPath().getList(".", ActionLog.class);
+         log.infof("[%d] Retrieved %d log entries", i, logs.size());
+         for (ActionLog l : logs) {
+            if (l.type == SlackChannelMessageAction.TYPE_SLACK_MESSAGE) {
+               log.infof("Found log for %d: %d, %s, %s: %s", l.testId, l.level, l.event, l.type, l.message);
+               errors++;
+            }
+         }
+      }
+      log.infof("Found %d errors for %s on test %d", errors, SlackChannelMessageAction.TYPE_SLACK_MESSAGE, -1);
+
+      // If only ...
+      // Assert.assertTrue("No slack action errors found", errors > 0);
```

Comment:
If we mock the slack service and use the mocked endpoint for testing, we can control the reponse, and therefore the errors we see in testing. A mocked service could be something as simple as ;

```
package io.hyperfoil.tools.horreum;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;
import org.eclipse.microprofile.openapi.annotations.tags.Tag;
import org.jboss.logging.Logger;

@ApplicationScoped
@Path("/api/chat.postMessage")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
@Tag(name = "SlackService", description = "Mock endpoint for Slack service.")
public class SlackDummyService {

   private static final Logger log = Logger.getLogger(SlackDummyService.class);

   @Inject
   ObjectMapper mapper;

   @POST
   public ObjectNode mockSlackEndpoint(String payload) {
      log.infof("Received payload: %s", payload);

      return mapper.createObjectNode().put("ok", true);
   }
}
```

And if we implement the config options as above, we can use that service (instead of the live slack service) with a config change in application.properties;

```
%test.horreum.action.slack.host=localhost
%test.horreum.action.slack.port=8081
%test.horreum.action.slack.ssl=false
```